### PR TITLE
Refactor : make code easier to understand

### PR DESCRIPTION
**Pull request by guilpier-code**
This PR is about refactoring the script that generates the reference results for end-to-end tests run in the CI.
This script was beginning to be difficult to understand

The code functionally does 4 things : 
- it searches for **Antares** studies in a batch directory and makes a list with the found studies
- it builds command to run of type : executable + options + study path as option
- it runs the previous command
- it move the results into a **reference** directory

**What was done** : 
- A tests environnement was prepared (golden master) to avoid coding error as much as possible.
- The 4 steps previously stated now appear clearly in the main program. Anything else was considered as implementation details and was moved to lower levels. So main program is shorter now.
-  Everything about making and running commands was gathered in a separate module (this code was previously scattered in different places).
- Still about making commands : the decision to make a command or another now relies on one parameter (as it should be) : **batch_name**. Previously several parameters were used to build a command.
- Functions and variables where renamed to be more meaningful

**Further possible improvements (draft / reminder)** : 
-  the command line arg **--thermal=xxxx** of binary **ts-generator** is completely retrieved from a file **clustersToGen.txt**.
    - do we need a new data file in the study for this ?
    - should we not remove the prefix **--thermal=** from this file, and keep only the **xxxx** part ?
-  wouldn't it be better to give the executable directly as argument of the script ? Instead we have to search for an exe in a certain path. Not sure we get value from this.
- There is not classes in this script, and therefore no polymorphism. This could allow to separate and factorize code more than it is now, paving a path to more changeability.
- About building commands : 
    - related code is gathered in a function. This function is long and could be split.
    - The command is totally re-built for each study path, although there is a common part to each command.

### CODE REVIEW

File: scripts/generate_reference.py [hunk 1]
```diff
@@ -33,35 +22,68 @@ def find_binary(path, binary_name):
...
+        command.append('--use-ortools')
+        command.append('--ortools-solver=' + opt_solver)
+    if named_mps_problems:
+        command.append('--named-mps-problems')
+    if ts_generator_path != "":
+        cluster_to_gen_file = open(study_path / "clustersToGen.txt", 'r')
+        cluster_to_gen = cluster_to_gen_file.readline().rstrip()  # remove new line char
+        cluster_to_gen_file.close()
+        command = [ts_generator_path, cluster_to_gen, str(study_path)]
+    return command
```
Comment (flomnes) [inline scripts/generate_reference.py:53]: Merge solver_path and ts_generator_path ?

One function for solver, one function for ts-generator ?

Comment (guilpier-code) [reply scripts/generate_reference.py:53]: > Merge solver_path and ts_generator_path ?
> 
> One function for solver, one function for ts-generator ?
It's a work in progress, first we isolate what belong together.
But we planed to do that

File: scripts/antares_test_utils.py [hunk 2]
```diff
@@ -14,16 +13,17 @@ def list_directories(directory):
             dir_list.append(x)
     return dir_list
 
-def list_studies(directory):
+def find_studies_in_batch_dir(batch_name):
+    batch_directory = Path(batch_name).resolve()
     studies = []
-    dir_path = Path(directory)
+    dir_path = Path(batch_directory)
```
Comment (a-zakir) [inline scripts/antares_test_utils.py:19]: dir_path == batch_directory ?

File: scripts/antares_test_utils.py [hunk 3]
```diff
@@ -44,51 +44,20 @@ def find_output_result_dir(output_dir):
...
-
-    enable_study_output(path, True)
-
-    result = launch_solver(solver_path,path, use_ortools, ortools_solver, named_mps_problems, ts_generator_path)
-
-    output_path = path / 'output'
+def move_output_to_reference(study_path):
+    output_path = study_path / 'output'
     list_dir = list_directories(output_path)
     assert len(list_dir) == 1
```
Comment (a-zakir) [inline scripts/antares_test_utils.py:56]: may add an explicit error message

File: scripts/antares_test_utils.py [hunk 4]
```diff
@@ -44,51 +44,20 @@ def find_output_result_dir(output_dir):
 def get_headers(df) -> set :
     return set(df.columns)
 
-def remove_outputs(study_path):
+def remove_possibly_remaining_outputs(study_path):
     output_path = study_path / 'output'
     files = glob.glob(str(output_path))
     for f in files:
         shutil.rmtree(f)
```
Comment (a-zakir) [inline scripts/antares_test_utils.py:52]:
```suggestion
    output_path = study_path / 'output'
   shutil.rmtree(output_path, ignore_errors=True)
```

File: scripts/generate_reference.py [hunk 5]
```diff
@@ -1,67 +1,35 @@
...
-from pathlib import Path
 
 import antares_test_utils as antares_utils
+from run_command_building import *
 import sys
 
+# Parsing args
 parser = argparse.ArgumentParser()
-parser.add_argument("root", help="Directory containing studies",
+parser.add_argument("batch_name", help="Btch directory (containing studies)",
```
Comment (a-zakir) [inline scripts/generate_reference.py:9]:
```suggestion
parser.add_argument("batch_name", help="Batch directory (containing studies)",
```

File: scripts/generate_reference.py [hunk 6]
```diff
@@ -1,67 +1,35 @@
...
                     type=str)
-
-parser.add_argument("solver", help="Path to antares-solver",
+parser.add_argument("path_where_to_find_exe", help="Path to binary to exe",
```
Comment (a-zakir) [inline scripts/generate_reference.py:11]: the option name ans it's description are not consistent

File: scripts/run_command_building.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        suffix=""
+
+    binary_path = Path(path)
+    if binary_path.is_file():
+        return path.resolve()
+    if binary_path.is_dir():
+        results = []
+        for path_item in binary_path.iterdir():
+            if path_item.is_file() and (f"antares-{exe_identifier}{suffix}" == path_item.name):
+                results.append(path_item)
```
Comment (a-zakir) [inline scripts/run_command_building.py:18]: break if the solver is found

File: scripts/run_command_building.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+def find_binary(path, exe_identifier):
+    if sys.platform.startswith("win"):
+        suffix=".exe"
+    else:
+        suffix=""
+
+    binary_path = Path(path)
+    if binary_path.is_file():
+        return path.resolve()
+    if binary_path.is_dir():
```
Comment (a-zakir) [inline scripts/run_command_building.py:14]: binary is not ubiquitous

File: scripts/antares_test_utils.py (hunk 2 above)
Comment (guilpier-code) [reply scripts/antares_test_utils.py]: I simplified this. 
Thanks for noticing it.

File: scripts/antares_test_utils.py (hunk 3 above)
Comment (guilpier-code) [reply scripts/antares_test_utils.py:55]: Done

File: scripts/antares_test_utils.py (hunk 4 above)
Comment (guilpier-code) [reply scripts/antares_test_utils.py]: Applied

File: scripts/generate_reference.py (hunk 5 above)
Comment (guilpier-code) [reply scripts/generate_reference.py:9]: Applied

File: scripts/run_command_building.py (hunk 7 above)
Comment (guilpier-code) [reply scripts/run_command_building.py]: I kind of applied your suggestion, and shrunk the code a bit more

**Review by a-zakir — APPROVED** [review]

File: scripts/generate_reference.py (hunk 6 above)
Comment (guilpier-code) [reply scripts/generate_reference.py]: corrected

**Review by flomnes — APPROVED** [review]

**Pull request merged**